src/System.IdentityModel.Tokens.Jwt/JwtConstants.cs

```diff
@@ -47,9 +47,9 @@ public static class JwtConstants
         public const string TokenTypeAlt = "urn:ietf:params:oauth:token-type:jwt";
 
         /// <summary>
-        /// Token format: 'header.payload.signature'. Signature is optional, but '.' is required.
+        /// Token format: 'header.payload.signature'. Signature is optional, and if missing, trailing '.' is also optional.
         /// </summary>
-        public const string JsonCompactSerializationRegex = @"^[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]*$";
+        public const string JsonCompactSerializationRegex = @"^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)\.(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)\.?(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$";
```

Comment:
I see what you are talking about...  The WSO2 API Gateway is sending the JWS in Base64 instead of Base64Url.  I will need to do a conversion, but then the existing regex at least passes the token.  Sorry for my confusion on that point.